src/configPolling.ts

```diff
@@ -0,0 +1,144 @@
+ *     much slower and is intended for the service to keep in sync with updates
+ *     to the configuration.
+ *
+ * It will automatically switch from the bootInternval to idleInterval after the
+ * first configuration module is loaded.
+ */
+export default class ConfigPoller {
+  /**
+   * The ConfigService the ConfigPoller is associated to
+   */
+  #config: ConfigService;
+  /**
+   * Configuration for polling
+   */
+  #settings: PollingConfig;
+  /**
+   * Tracks reference handle of the setInterval
+   */
+  #interval: NodeJS.Timeout | null = null;
+
+  constructor(config: ConfigService, settings: PollingConfig) {
+    this.#config = config;
+    this.#settings = settings;
+  }
+
+  /**
+   * The polling interval is dependant on if we've already successfully loaded a
+   * configuration module or not.
+   */
+  get pollInterval() {
+    return this.#config.isLoaded
+      ? this.#settings.idleInterval
+      : this.#settings.bootInternval;
+  }
+
+  /**
+   * Fired when the polling interval ticks
+   */
+  #pollTick = async () => {
+    logger.debug('Polling tick...');
+
+    // Load deadline should match the polling interval
+    const loadDeadline = this.pollInterval;
+
+    let config: any = null;
+
+    try {
+      config = await this.#config.fetchConfig(loadDeadline);
+    } catch (error) {
+      const didAbort = error instanceof fetch.AbortError;
+
+      if (didAbort) {
+        logger.debug('Config resolution aborted for next polling tick');
+      } else {
+        logger.warn(`Failed to resolve config while polling with error: ${error}`);
+      }
+
+      return;
+    }
+
+    const [validConfig, errors] = validateConfig(config);
+
+    if (errors !== undefined) {
+      logger.warn(
+        this.#config.isLoaded
+          ? 'Resolved NEW but INVALID config. Not updating...'
+          : 'Resolved INVALID config during boot! Trying again in XXXXXXXXX'
+      );
+
+      // TODO(epurkhiser): It's likely we need to yell a little louder here
+      // if we've loaded an invalid configuration.
+
+      return;
+    }
+
+    if (validConfig.version === this.#config.version) {
+      logger.debug('Resolved the SAME configuration via polling. Nothing to do.');
+      return;
+    }
+
+    logger.info(
+      `Resolved new config via polling: ${validConfig.renderConfig.size} styles available.`,
+      {version: validConfig.version}
+    );
+
+    const wasBootPolling = !this.#config.isLoaded;
+
+    // Successful configuration update!
+    this.#config.setVersion(validConfig.version);
+    this.#config.setRenderConfig(validConfig.renderConfig);
+
+    // Switch to idle polling,
+    if (wasBootPolling) {
+      logger.info('Config polling switching to idle mode');
+
+      this.stopPolling();
+      this.startPolling({immediate: false});
+    }
+  };
+
+  /**
+   * Poll to load + update the specified configuration module
+   */
+  startPolling({immediate}: StartOptions) {
```

Comment:
It's less important since it's backend instead of frontend, so it's less likely to cause thundering herd, but maybe still consider adding some sort of backoff to the poll to not spam during an outage? Not blocking, just a suggestion.

Reply:
I think we _should be_ okay for now.

Reply:
Agree with the back-off comment. Would be great if we track that as a TODO in the code or as an issue on the repo.